File: src/models/template.js
// src/models/template.js

const { DataTypes } = require('sequelize');
const { sequelize } = require('../database/db');

const Template = sequelize.define('Template', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  googleDocId: {
    type: DataTypes.STRING(255),
    allowNull: false,
    field: 'google_doc_id'
  },
  contentPreview: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'content_preview'
  },
  isActive: {
    type: DataTypes.BOOLEAN,
    defaultValue: true,
    field: 'is_active'
  },
  createdAt: {
    type: DataTypes.DATE,
    field: 'created_at'
  },
  updatedAt: {
    type: DataTypes.DATE,
    field: 'updated_at'
  }
}, {
  tableName: 'templates',
  timestamps: true,
  underscored: true
});

module.exports = Template;

File: mobile/lib/models/template.dart
// lib/models/template.dart

class Template {
  final int id;
  final String name;
  final String? description;
  final String? googleDocId;
  final String? contentPreview;
  final bool isActive;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Template({
    required this.id,
    required this.name,
    this.description,
    this.googleDocId,
    this.contentPreview,
    required this.isActive,
    this.createdAt,
    this.updatedAt,
  });

  factory Template.fromJson(Map<String, dynamic> json) {
    return Template(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      googleDocId: json['googleDocId'] as String?,
      contentPreview: json['contentPreview'] as String?,
      isActive: json['isActive'] as bool? ?? true,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'] as String)
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'googleDocId': googleDocId,
      'contentPreview': contentPreview,
      'isActive': isActive,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}

